### PR TITLE
Flag clearcut through environment variable

### PR DESCRIPTION
**Pull request by elharo**
Also log non-200 responses
@briandealwis

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java [hunk 1]
```diff
@@ -259,6 +259,14 @@ private void sendPing(PingEvent pingEvent) {
     }
   }
 
+  private static boolean useClearCut() {
+    String s = System.getenv("USE_CLEARCUT");
+    if (s == null) {
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: nit: with `"true".equalsIgnoreCase(s)`, the null check is unnecessary.

File: plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java [hunk 2]
```diff
@@ -247,10 +244,13 @@ private void sendPing(PingEvent pingEvent) {
         }
       }
       
-      if (USE_CLEAR_CUT) {
+      if (useClearCut()) {
         try {
           String json = jsonEncode(pingEvent);
-          HttpUtil.sendPost(clearCutUrl, json, "application/json");
+          int resultCode = HttpUtil.sendPost(clearCutUrl, json, "application/json");
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: So I noticed that _google-http-client_'s `HttpRequest#execute()`, which is used under the hood here, may throw an `HttpResponseException` for an error response when `HttpRequest#getThrowExceptionsOnExecuteError()` is true, which defaults to true.

File: plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java [hunk 3]
```diff
@@ -259,6 +259,14 @@ private void sendPing(PingEvent pingEvent) {
     }
   }
 
+  private static boolean useClearCut() {
+    String s = System.getenv("USE_CLEARCUT");
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: Are single letters allowed now?

File: plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java (hunk 2 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: …which means this logging should be put below, I suspect.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: I think we're using java.net.URLConnection, not HttpRequest, here.

Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: No it's using google-http-client.

Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: I have to say it is not using Google HTTP Client. It is used only for the multi-part post that is not being used.

Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java]: Sorry I was looking at `sendPostMultipart` :blush:

**Pull request merged**